static/js/hotkey.js

```diff
@@ -752,25 +752,25 @@ export function process_hotkey(e, hotkey) {
             return true;
     }
 
-    // We don't want hotkeys below this to work when recent topics is
-    // open. These involve compose box hotkeys and hotkeys that can only
-    // be done performed on a message.
-    if (recent_topics.is_visible()) {
-        return true;
-    }
-
+    const opts = {trigger: "compose_hotkey"};
     // Compose box hotkeys
     switch (event_name) {
         case "compose": // 'c': compose
-            compose_actions.start("stream", {trigger: "compose_hotkey"});
+            if (recent_topics.is_visible()) {
+                opts.stream = recent_topics.get_focused_stream();
+            }
+            compose_actions.start("stream", opts);
```

Comment:
Would it make sense to modify `fill_in_opts_from_current_narrowed_view` to do this behavior instead?

(I'm thinking about a future world where we have a "Recent topics" like view that includes PMs, and how we'd want that to work).